src/redis-cli.c

```diff
@@ -5864,6 +5865,29 @@ static int clusterManagerCommandAddNode(int argc, char **argv) {
     listAddNodeTail(cluster_manager.nodes, new_node);
     added = 1;
 
+    if (!master_node) {
+        /* Send functions to the new node, if new node is a replica it will get the functions from its primary. */
+        clusterManagerLogInfo(">>> Getting functions from cluster\n");
+        reply = CLUSTER_MANAGER_COMMAND(refnode, "FUNCTION DUMP");
+        if (!clusterManagerCheckRedisReply(refnode, reply, &err)) {
+            clusterManagerLogInfo(">>> Failed retrieving Functions from the cluster, "
+                    "skip this step as Redis version do not support function command (error = '%s')\n", err? err : "NULL reply");
+            if (err) zfree(err);
+        } else {
+            assert(reply->type == REDIS_REPLY_STRING);
+            clusterManagerLogInfo(">>> Send FUNCTION RESTORE to %s:%d\n", ip, port);
+            function_restore_reply = CLUSTER_MANAGER_COMMAND(new_node, "FUNCTION RESTORE %b APPEND", reply->str, reply->len);
```

Comment:
@madolson so you suggest to use `FLUSH` and check before that there is not existing functions?

Reply:
Actually it will be enough to just verify that there is no functions in the new node...

Reply:
Fixed and update to comment.

Reply:
i think adding another round trip to check if the node has functions may be excessive and more complicated.
but one thing's for sure, we do wanna either override the conflicting ones, or fail the add-node (which i think is better).
so the disadvantage here (when using APPEND, and failing if there are conflicts) is that there's a chance the new node will have some additional functions that don't exist in the other nodes.
but there's actually no guarantee that other nodes don't have such mismatches (if the user uploaded different functions to each).

so maybe the question is where this new node come from, and if it was wiped clean, and re-configured before being added.
@madolson is there some mechanism for that? or common practice?
if not, then maybe we do need FLUSH

Reply:
ohh, i see it was already implemented, and looks like not a lot of code.
so the only possible concern is that maybe we're overriding some data that the user didn't realize that he's gonna lose.
but that's also true about data in the keyspace, right? something in that process is calling FLUSHALL?

Reply:
@oranagra why we overriding data? If there is any function on the new node we failed the process, otherwise we add the node and the functions (we are not overriding because we know there are no functions on the new node).

Reply:
right. sorry.. i guess i mixing the case using FLUSH without a check in my head.

Reply:
Sorry for being slow, doing the check outlined here seems like the safest approach, so LGTM.